src/ui/components/modals/report_modal.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{
+  serverfns::{comments::create_report_comment_action, posts::create_report_post_action},
+  ui::components::common::{
+    icon::{Icon, IconType},
+    text_input::TextInput,
+  },
+  utils::{
+    create_user_apub_name,
+    types::{ContentActionType, ReportModalData},
+  },
+};
+use html::Dialog;
+use leptos::*;
+use leptos_router::ActionForm;
+
+#[component]
+fn ReportForm(
+  id: Signal<i32>,
+  creator_actor_id: Signal<String>,
+  content_action_type: Signal<ContentActionType>,
```

Comment:
Same ID stuff, you could use a PostOrComment enum to differentiate,

Reply:
Done